main.go

```diff
@@ -53,6 +54,7 @@ func (c *rootCmd) RegisterFlags(fl *flag.FlagSet) {
 	fl.BoolVar(&c.skipSync, "skipsync", false, "skip syncing local settings and extensions to remote host")
 	fl.BoolVar(&c.syncBack, "b", false, "sync extensions back on termination")
 	fl.BoolVar(&c.printVersion, "version", false, "print version information and exit")
+	fl.BoolVar(&c.noReuseConnection, "no-reuse-connection", false, "do not reuse SSH connection via control socket")
```

Comment:
I'm biased since I requested this feature, but I think that it should be opt-out and the default should be reusing the `ssh` connection. I don't see a benefit to not making this feature a default.

Reply:
I think it's more user friendly to have this enabled by default. If it fails to create the master connection, it'll safely fall back to doing it the normal way.

Reply:
Ok with me